src/disk.hpp

```diff
@@ -175,7 +175,11 @@ struct FileDisk {
                 std::cout << "Only read " << amtread << " of " << length << " bytes at offset "
                           << begin << " from " << filename_ << "with length " << writeMax
                           << ". Error " << ferror(f_) << ". Retrying in five minutes." << std::endl;
+                // Close, Reopen, and re-seek the file
```

Comment:
I think someone reading this code a year from now may wonder why we close and re-open the file again. I think this is a good place for a comment explaining the problem this solved.